README.md

```diff
@@ -1,49 +1,146 @@
 
+## Usage
 
-## Build Instructions
+#### Building & booting the environment
+You can get started with *Fuzzy* quite simply, just launch a
+[Docker](https://www.docker.com/) container as follows:
+
+```
+docker run -ti verificarlo/fuzzy
+```
+
+If you're on a shared system and would prefer to use [Singularity](https://sylabs.io/),
+that's no problem, just convert the container using the appropriate method for
+your system ([e.g.](https://docs.computecanada.ca/wiki/Singularity#Creating_images)).
+
+If you would like to build the environment locally on your system, look at the
+Dockerfiles in `docker/base/` to see how installation was performed.
+
+#### Adding your software
+If your code is written in, say, Python, then you can simply mount your code to
+this environment, run it using the contained interpreter, and experince the *:fuzz:*.
+If your code loads shared libraries, such as `libmath` or `lapack`, make sure that
+they are using the instrumented versions of these libraries made available in
+`/usr/local/lib/`. You can check whether this is the case using `ldd </path/to/yourbinary>`.
+
+The next step is to make sure the environment is configured to introduce perturbations
+the way you expect. You can start with a configuration which performs perturbations
+akin to randomizing machine error with the following:
+
+```
+echo "libinterflop_mca.so -m mca --precision-binary32=24 --precision-binary64=53" > $VFC_BACKENDS_FROM_FILE
+```
+
+For more usage instructions on how to control how perturbations are introduced
+and where they occur within operations, please refer to the
+[Verificarlo](https://github.com/verificarlo/verificarlo) repository.
+
+A simple sanity check to verify that your code is using the perturbed libraries
+is to dramatically reduce the precision in your configuration by changing the
+`precision-binary32` and `precision-binary64` values to something small (e.g. 1),
+boot up a debugging session in your environment (e.g. a Python shell, GDB, etc.),
+load a math library, and run a basic math function a few times (e.g. `sin(1)`) —
+you should get different results if your instrumentation is setup properly.
+***Important***: Don't forget to set the precision back to your desired level
+prior to performing your experiments!
+
+#### Running Fuzzy workflows
+In the context of *fuzzy* experiments, it is important to remember that by default
+each execution will be evaluated with a unique random state, meaning that when you
+run it again you may get slightly (or very) different results. If your goal is to
+characterize the variability in your results, or obtain a robust estimate of the
+"true" mean answer, you will need to then run your tool multiple times and compare
+each execution.
+
+It's also important to remember that the execution time will be increased when using
+*Fuzzy*, as compared to running tools in a determinitic environment. Depending on
+the tool, the instrumentation, and the MCA mode, this additional overhead may range
```

Comment:
MCA is "Monte-Carlo Arithmetic"? PB is "Protocol Buffers"? I might be missing it somewhere, but I don't see what the options for the perturbation modes are

Reply:
Thanks for pointing out. Options are available at the [verificarlo](https://github.com/verificarlo/verificarlo/blob/master/README.md#mca-backend-libinterflop_mcaso) project. We should add a section for them here.

Reply:
To answer MCA stands for Monte Carlo Arithmetic. In this stochastic arithmetic you have three modes for perturbing your floating-point operations `x = a op b` where op in {+,-,*,/}: 
- Random Rounding (RR): `x = inexact(a op b)`
- Precision Bounding (PB): `x = inexact(a) op inexact(b)`
- Full MCA (MCA): `x = inexact(inexact(a) op inexact(b))`